src/log/iobuf.rs

```diff
@@ -258,14 +258,15 @@ impl IoBufs {
             let res_start = buf_offset as usize;
             let res_end = res_start + buf.len();
             let destination = &mut (out_buf)[res_start..res_end];
+            let reservation_offset = log_offset + buf_offset as LogID;
 
             return Reservation {
                 idx: idx,
                 iobufs: self,
                 data: buf,
                 destination: destination,
                 flushed: false,
-                base_disk_offset: log_offset as LogID,
+                reservation_offset: reservation_offset,
```

Comment:
this was a stupid bug that went undetected because the pagecache wasn't well tested before, where the log id for a reservation was just the ID of the base disk offset. it's amazing that the tree was working with recovery tests...

Reply:
😃 Nice catch 👍!